internal/store/templateversion.go

```diff
@@ -0,0 +1,157 @@
+package store
+
+import (
+	"context"
+	b64 "encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/store/model"
+	"github.com/flightctl/flightctl/internal/util"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+type TemplateVersion interface {
+	Create(ctx context.Context, orgId uuid.UUID, templateVersion *api.TemplateVersion) (*api.TemplateVersion, error)
+	List(ctx context.Context, orgId uuid.UUID, listParams ListParams) (*api.TemplateVersionList, error)
+	DeleteAll(ctx context.Context, orgId uuid.UUID, fleet *string) error
+	Get(ctx context.Context, orgId uuid.UUID, fleet string, name string) (*api.TemplateVersion, error)
+	Delete(ctx context.Context, orgId uuid.UUID, fleet string, name string) error
+	InitialMigration() error
+}
```

Comment:
these interfaces feel like copy/paste boilerplate I wonder if we can use go generics here. High level we have 2 objects

- Resource
- ResourceList

So we have a single interface implemented for each resource, this can be a followup but I would like to track it.
```
type Service[T Resource, L ResourceList[T]] interface {
	Create(ctx context.Context, orgId uuid.UUID, resource *T) (*T, error)
	List(ctx context.Context, orgId uuid.UUID, listParams ListParams) (*L, error)
	DeleteAll(ctx context.Context, orgId uuid.UUID, fleet *string) error
	Get(ctx context.Context, orgId uuid.UUID, fleet string, name string) (*T, error)
	Delete(ctx context.Context, orgId uuid.UUID, fleet string, name string) error
	InitialMigration() error
}
```

Reply:
I would love to see that happen.  There is a ton of copy-paste in this service for each resource.  Definitely not in the scope of this PR though.  @sagidayan was looking into this.

Reply:
100% followup